nixos/modules/services/monitoring/prometheus/exporters/restic.nix

```
@@ -0,0 +1,114 @@
+{ config, lib, pkgs, options }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.restic;
+in
+{
+  port = 9753;
+  extraOpts = {
+    repository = mkOption {
+      type = types.str;
+      description = lib.mdDoc ''
+        repository to monitor.
+      '';
+      example = "sftp:backup@192.168.1.100:/backups/example";
+    };
+
+    passwordFile = mkOption {
+      type = types.str;
+      description = lib.mdDoc ''
+        Read the repository password from a file.
+      '';
+      example = "/etc/nixos/restic-password";
+    };
+
+    environmentFile = mkOption {
+      type = with types; nullOr str;
+      default = null;
+      description = lib.mdDoc ''
+        file containing the credentials to access the repository, in the
+        format of an EnvironmentFile as described by systemd.exec(5)
+      '';
+    };
+
+    rcloneOptions = mkOption {
+      type = with types; nullOr (attrsOf (oneOf [ str bool ]));
+      default = null;
+      description = lib.mdDoc ''
+        Options to pass to rclone to control its behavior.
+        See <https://rclone.org/docs/#options> for
+        available options. When specifying option names, strip the
+        leading `--`. To set a flag such as
+        `--drive-use-trash`, which does not take a value,
+        set the value to the Boolean `true`.
+      '';
+    };
+
+    rcloneConfig = mkOption {
+      type = with types; nullOr (attrsOf (oneOf [ str bool ]));
+      default = null;
```

The `null` doesn't seem to be a deciding factor in anything other than what to merge in the config. Maybe we could reduce some complexity by changing the type of this to non-nullable attrs with default `{ }`, and merging unconditionally?

```suggestion
    rcloneOptions = mkOption {
      type = with types; attrsOf (oneOf [ str bool ]);
      default = { };
      description = lib.mdDoc ''
        Options to pass to rclone to control its behavior.
        See <https://rclone.org/docs/#options> for
        available options. When specifying option names, strip the
        leading `--`. To set a flag such as
        `--drive-use-trash`, which does not take a value,
        set the value to the Boolean `true`.
      '';
    };

    rcloneConfig = mkOption {
      type = with types; attrsOf (oneOf [ str bool ]);
      default = { };
```